webpay/pay/tests/test_tasks.py

```diff
@@ -954,6 +954,17 @@ def test_prevent_reconfiguring_transaction(self):
         eq_(configured, False)  # Second call should do nothing.
         eq_(self.start_pay.call_count, 1)
 
+    def test_always_reconfigure_transaction_after_error(self):
```

Comment:
there is a [test for that](https://github.com/kumar303/webpay/blob/retry/webpay/pay/tests/test_tasks.py#L954) but it doesn't check the session directly. It tests the behavior; it executes the helper in a successful state and re-executes the helper to make sure it doesn't start the task again (this implicitly tests the session but doesn't rely on it as an implementation).